### ISSUE TITLE
Suggestion for what happens when you upload a grain backup of an app that isn't installed

### ISSUE THREAD

**Issue by jadeqwang** [sandstorm-io/sandstorm#1107]
Instead of this sequence: "upload failed" (discard upload) -> go to app market -> install relevant app -> re-upload grain backup -> use backed up grain

I suggest this sequence: "app not installed" (keep uploade) -> go to app market -> install relevant app -> use backed up grain grain

The thing I ran into was for a small-ish TextEditor grain, and would have been a giant pain if it were some huge file in a FileDrop.

**Comment by ocdtrekkie** [comment]
Agreed for sure. Honestly, I feel like if you've uploaded the grain, and the server can find the relevant app package to open it, it should just ask if it's okay to get the app from the store and pull it directly.

**Comment by jadeqwang** [comment]
Yep, agree with @ocdtrekkie. Ideal sequence would probably be: 
"app not installed" (keep upload) -> "app exists. ok to install?" yes -> use backed up grain
